Add str.replace() method (#8489)
mesonbuild/meson: docs/markdown/Syntax.md

Strings also support a number of other methods that return transformed
copies.

#### .replace()

> Since 0.58.0, you can now replace characters in a string.

Reviewer: ```suggestion
Since 0.58.0, you can replace a substring from a string.
```
To make it clear it's not replacing characters, but substring.

```meson

> # Similar to the Python str.replace(). Replaces instances of one character with another

Reviewer: I think we voluntarily avoid documenting stuff as reference to python, because it's really an implementation detail.

Reviewer: I would also specify "replaces **all** instances". (I hate how this doc is duplicated)

s = 'semicolons;as;separators'

> s = s.replace(';', ' ')

Reviewer: I would rather use an example that makes it cleared it's replacing a substring and not characters.
